### PR TITLE
sanitize notification strings

### LINKED ISSUE #6: possible exploit with Notifications on Windows
https://twitter.com/3xp0rtblog/status/1661826845791117314

https://twitter.com/vxunderground/status/1661789069670780934

image

image

### PR DESCRIPTION
**Pull request by zoff99**

### CODE REVIEW

File: src/platform/desktop_notifications/desktopnotify.h [hunk 1]
```diff
@@ -35,6 +35,7 @@ class DesktopNotify : public QObject
     Q_OBJECT
 public:
     explicit DesktopNotify(Settings& settings);
+    QString sanitize_text_for_notifications(const QString& input_text);
```
Comment (Green-Sky) [inline src/platform/desktop_notifications/desktopnotify.h]: rename to `sanitizeTextForNotifications` (camelcase) to better fit in.

File: src/platform/desktop_notifications/desktopnotify.cpp [hunk 2]
```diff
@@ -43,14 +44,24 @@ DesktopNotify::DesktopNotify(Settings& settings_)
     connect(&notifyCore, &Snore::SnoreCore::notificationClosed, this, &DesktopNotify::onNotificationClose);
 }
 
+QString DesktopNotify::sanitize_text_for_notifications(const QString& input_text)
+{
+    QString output_text = QString(input_text);
+    output_text.replace(QRegularExpression("[^a-zA-Z0-9 _:]"), "_"); // allow only a A-Z and 0-9 and "_" and " " and ":" chars
```
Comment (Green-Sky) [inline src/platform/desktop_notifications/desktopnotify.cpp:50]: this will destroy notifications for non-latin-script languages. but looks safe.

**Review by Green-Sky — CHANGES_REQUESTED** [review]

File: src/platform/desktop_notifications/desktopnotify.cpp (hunk 2 above)
Comment (zoff99) [reply src/platform/desktop_notifications/desktopnotify.cpp:50]: we need to be strict with giving random strings to commandline programs

File: src/platform/desktop_notifications/desktopnotify.h (hunk 1 above)
Comment (zoff99) [reply src/platform/desktop_notifications/desktopnotify.h]: done

**Review by Green-Sky — APPROVED** [review]

**Comment by GrayHatter** [comment]
thank you @zoff99

**Pull request merged**